manager/logbroker/broker.go

```diff
@@ -301,6 +330,19 @@ func (lb *LogBroker) PublishLogs(stream api.LogBroker_PublishLogsServer) error {
 			return grpc.Errorf(codes.InvalidArgument, "missing subscription ID")
 		}
 
+		if currentSubscription == nil {
+			{
```

Comment:
I still don't get it. `return` triggers the defer, and I can't understand why an extra level of scope would make a difference.